Functions/InputOutput/xASL_io_ReadDataPar.m

```diff
@@ -147,37 +147,56 @@
 	else
 		x = jsonData;
 	end
-    
-	sFields = fieldnames(x);
-	for n=1:size(sFields,1)
-		% Check if the current field is valid (char, numeric value, structure or cell array)
-		if ~(ischar(x.(sFields{n})) || isstruct(x.(sFields{n})) || isnumeric(x.(sFields{n})) || islogical(x.(sFields{n})) || iscell(x.(sFields{n})))
-			fprintf('\n%s\n',char(sFields{n}));
-			warning('JSON field type could be invalid...');
+	if isfield(x,'StudyPars')
+		bStudyPars = 1;
+		nContext = length(x.StudyPars);
+	else
+		bStudyPars = 0;
+		nContext = 1;
+	end
+	
+	for iSubfield = 1:nContext
+		if bStudyPars
+			subx = x.StudyPars{iSubfield};
+		else
+			subx = x;
 		end
-		if strcmpi(sFields{n},'group') % Convert group fields to correct Matlab cell arrays
-			% Generate new Matlab cell array
-			x.group{str2num(strrep(sFields{n},'group',''))} = x.(sFields{n});
-			% Remove old field
-			x = rmfield(x,sFields{n});
-		elseif strcmpi(sFields{n},'LOAD') % Handle load commands
-			loadPaths = fieldnames(x.(sFields{n}));
-			for m=1:size(loadPaths,1)
-				% Don't forget to use \\ instead of \ in your paths
-				if exist(x.(sFields{n}).(loadPaths{m}),'file')
-					load(fullfile(x.(sFields{n}).(loadPaths{m})));
+		sFields = fieldnames(subx);
+		for n=1:size(sFields,1)
```

Comment:
please let's not use single-letter variables. subx m n should all be readable words

Reply:
subx = sub-x. x is already a one letter variable.
m and n replaced.

Reply:
OK then please call it `subjectX` and/or clarify in the header what subx means

Reply:
`subfieldX` - clarified in the comment, not in the header, as this is only a local variable.

Reply:
Great, good example that abbreviating costs time. I would have needed to figure out that `sub` didn't refer to `subject` but to `subfield`. Writing this out saves time.

Please resolve.